libs/contrib/Data/Nat/Factor.idr

```diff
@@ -124,8 +123,7 @@ multOneSoleNeutral (S k) (S (S j)) prf =
         rewrite plusCommutative k j in
         rewrite sym $ plusAssociative j k (k * S j) in
         rewrite sym $ multRightSuccPlus k (S j) in
-        succInjective k (j + (S (S (j + (k * (S (S j))))))) $
-        succInjective (S k) (S (j + (S (S (j + (k * (S (S j)))))))) prf
+        inj S $ injective prf
```

Comment:
I think it would be clearer if this were two direct calls to `injective`, rather than one to the `inj` wrapper.

Reply:
The `inj` wrapper is necessary because `f` cannot be inferred here. We could pass it implicity, but this is exactly what the `inj` wrapper is for